### ISSUE TITLE
Unable to read valid PNG

### ISSUE THREAD

**Issue by Anuken** [treeform/pixie#175]
The following image cannot be read:

![invalid](https://user-images.githubusercontent.com/10100000/117575707-0dabe080-b0b1-11eb-877b-66ade3e068a7.png)

```
$ file invalid.png 
invalid.png: PNG image data, 96 x 96, 8-bit/color RGBA, non-interlaced
```

```nim
import pixie
let img = readImage("invalid.png")
```

->

```
test.nim(2) test
pixie-1.1.3/pixie.nim(38) readImage
pixie-1.1.3/pixie.nim(23) decodeImage
pixie-1.1.3/pixie/fileformats/png.nim(430) decodePng
pixie-1.1.3/pixie/fileformats/png.nim(349) decodePng
Error: unhandled exception: Invalid PNG buffer, unable to load [PixieError]
```

Pixie version: 1.1.3
Nim Compiler Version 1.4.6 [Linux: amd64]

**Comment by treeform** [comment]
How did you generate the file? It appears to have extra NULL character at the end. Maybe some sort of null-terminated-string copy? It does not conform to spec but it looks like most png editors/viewers ignore that 0 byte at the end. Maybe we should too?

![image](https://user-images.githubusercontent.com/125948/117577139-6c675f00-b09d-11eb-9f85-735879650372.png)

**Comment by Anuken** [comment]
> How did you generate the file?

This one wasn't made by me, so I'm not sure. I'm guessing a defective image editor added it in somewhere.

For what it's worth, Aseprite, GIMP and my system image viewer are all able to load it, so making Pixie accept trailing data is probably a good idea.

**Comment by guzba** [comment]
This was fixed here: https://github.com/treeform/pixie/pull/176/commits/8af4f9d8c1807ae13bdc51835ac666ef85f9d2a3

This fix is in a dev branch right now but will be merged into master and a tagged release soon. We're finishing up the branch in the next few days hopefully.

**Comment by Anuken** [comment]
Thanks!

**Comment by guzba** [comment]
This fix is included in release 1.2.0 which is live as of a day or two ago.
